### PR TITLE
Updated player.ready to always be async

### PR DESCRIPTION
**Pull request by heff**
Addresses #1667 

I've changed ready() and triggerReady() to always be async. However now a bunch of track tests are falling. 

@gkatsev, can you provide any insight here? I would guess it has to do with the text track display.
https://github.com/videojs/video.js/blob/1ff361a825400a2be803d6d4f0b72e0ecfbc8e6f/src/js/tracks/text-track-display.js#L41
```
Text Track Controls should be displayed when text tracks list is not empty FAILED
  control is displayed
  textTracks contains one item. Expected: 1 Actual: 0
```

If I move the addRemoteTextTrack part out of the ready listener the tests actually pass, but then other tracks tests fail.
```
Tracks TextTrackDisplay initializes tracks on player ready FAILED
  TypeError: Cannot read property 'tracks' of undefined
```

### CODE REVIEW

**Comment by gkatsev** [comment]
It probably has to do with the workarounds I've done previously to get around the fact that html5 is sync and flash is async.

**Comment by gkatsev** [comment]
Also, code-wise, LGTM

File: src/js/component.js [hunk 1]
```diff
@@ -730,22 +731,30 @@ class Component {
...
+    this.setTimeout(function(){
+      let readyQueue = this.readyQueue_;
 
-    if (readyQueue && readyQueue.length > 0) {
+      if (readyQueue && readyQueue.length > 0) {
+        readyQueue.forEach(function(fn){
+          fn.call(this);
+        }, this);
 
-      for (let i = 0; i < readyQueue.length; i++) {
```
Comment (gkatsev) [inline src/js/component.js]: kill those for-loops with extreme prejudice! :D

**Comment by gkatsev** [comment]
Also, didn't someone say that text tracks are generally broken in master, currently? Possibly related to that as well.

**Comment by gkatsev** [comment]
It's possible that we can remove the `player.ready()` wrapped for tracks and then just remove the test that checks that tracks are created on `ready`. I think that was one of my workarounds for this.

**Comment by gkatsev** [comment]
Ah, I know what the fix for the tests is. We need to actually trigger ready by ticking the clock forward.

**Comment by gkatsev** [comment]
I have a PR against this that fixes the text track errors: https://github.com/heff/video.js/pull/9

**Comment by heff** [comment]
Thanks! Nice use of the PR PR. :)

I tried that with one test but only ticked 1ms, which apparently isn't enough even though ready() waits 1ms. Guess it's safer to be more liberal with the time.

Tests are passing now. Attempting to remove the ready listener now.

**Comment by heff** [comment]
Removed it and it's just the one test breaking now that tests specifically for that (like you said). Killing that test.

File: src/js/component.js [hunk 2]
```diff
@@ -730,22 +731,30 @@ class Component {
    * @return {Component}
    */
   triggerReady() {
+
+    // // Ready should only be fired once
```
Comment (heff) [inline src/js/component.js]: @dmlap, I had this change in there but then decided to hold on it until we figured out how we wanted to handle tech changes. I forget where we landed, but also realized that the readyQueue does reset itself after ready is triggered, so ready listeners won't fire again on a tech change. That is, unless you listen for the ready event, not pass the function to ready(fn).

But basically the ready event was meant to let you add new ready listeners when the tech changes, it just may not be so clear that's the purpose. Maybe a `techloaded` event is the answer.

File: src/js/component.js (hunk 1 above)
Comment (heff) [inline src/js/component.js]: :curly_loop: :gun: :scream_cat:

**Pull request merged**